src/xma/include/lib/xmahw.h

```diff
@@ -47,12 +49,23 @@ typedef struct XmaHwKernel
     uint32_t    ddr_bank;
     //For execbo:
     int32_t     kernel_complete_count;
+    std::unique_ptr<std::condition_variable> kernel_complete_count_cv;
+    std::unique_ptr<std::atomic<bool>> kernel_complete_count_locked;
+
     void*       kernel_cmd_queue;
     void*       kernel_cmd_completion_queue;
     uint32_t    kernel_execbo_handle[MAX_EXECBO_POOL_SIZE];
     char*       kernel_execbo_data[MAX_EXECBO_POOL_SIZE];
     bool        kernel_execbo_inuse[MAX_EXECBO_POOL_SIZE];
     uint32_t    reserved[16];
+
+    XmaHwKernel(): kernel_complete_count_cv(new std::condition_variable),
```

Comment:
Is there any reason why this is heap allocated?

Reply:
I will remove condition_variable

Reply:
changed